pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -87,6 +88,10 @@
   bpftools,
   libbpf,
 
+  # vmlinux.h file containing the kernel's BTF information,
+  # used throughout systemd's BPF programs.
+  vmlinux-h ? "${callPackage ./vmlinux-btf.nix { }}/vmlinux.h",
```

Comment:
We might end up needing this for packages that use bpf CO-RE other than systemd, right? Why not make it a top-level attribute?

Reply:
The configs selected are systemd-specific. This is also the reason https://github.com/libbpf/vmlinux.h wouldn't work --- it doesn't have enough configs enabled.

Although... I guess it can't really hurt to have one central kitchen sink `vmlinux.h` in Nixpkgs. What do you think?

Reply:
I don't think this would need to happen in this PR? These files are inherently use-case-specific.

Reply:
> Although... I guess it can't really hurt to have one central kitchen sink `vmlinux.h` in Nixpkgs. What do you think?

I think this would make much more sense.

Reply:
- Sorry for the bad naming in the example: it builds the vmlinux ELF. I needed vmlinux to generate vmlinux.h later, with `bpftool`. Same as you do, I think.

- Maybe, but at least a user can just supply a kernel binary, rather than the sources to generate this infos. It seems more convenient to me.